infra/terraform/modules/repositories/main.tf

```diff
@@ -61,6 +70,15 @@ resource "github_repository_collaborator" "owners" {
   permission = "maintain"
 }
 
+resource "github_team_repository" "groups" {
+  for_each = {
+    for v in local.groups : "${v.repo}/${v.groups}" => v
+  }
+  repository = each.value.repo
+  team_id    = each.value.group
```

Comment:
Yes - Although I'd prefer to avoid adding more data-sources if at all possible given the already large number of API requests. Long term we might move them all to a dedicated module to reduce/deduplicate the number of data calls.